File: NeuroflowLib/supervised_sample_entry.cpp
#include "stdafx.h"
#include "supervised_sample_entry.h"
#include "data_array.h"

USING;

supervised_sample_entry::supervised_sample_entry(const data_array_ptr& input) : data(input, null, null) 
{ 
    verify_arg(input != null, "input");
}

supervised_sample_entry::supervised_sample_entry(const data_array_ptr& input, const data_array_ptr& desiredOutput, const data_array_ptr& actualOutput) :
data(input, desiredOutput, actualOutput)
{
    verify_arg(input != null, "input");
    verify_arg(desiredOutput != null, "desiredOutput");
    verify_arg(actualOutput != null, "actualOutput");
    verify_arg(desiredOutput->size() == actualOutput->size(), "desiredOutput, actualOutput");
}

const data_array_ptr& supervised_sample_entry::input() const
{
    return get<0>(data);
}

const data_array_ptr& supervised_sample_entry::desired_output() const
{
    return get<1>(data);
}

const data_array_ptr& supervised_sample_entry::actual_output() const
{
    return get<2>(data);
}

bool supervised_sample_entry::has_output() const
{
    return desired_output() != null && actual_output() != null;
}

File: _vault/NFBak/NFBak/Neuroflow_vsonline/Neuroflow/Neuroflow-Reborn/Neuroflow/Data/SupervisedSampleEntry.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Neuroflow.Data
{
    public sealed class SupervisedSampleEntry : IDisposable
    {
        public SupervisedSampleEntry(DataArray input)
        {
            Args.Requires(() => input, () => input != null);

            Input = input;
        }

        public SupervisedSampleEntry(DataArray input, DataArray desiredOutput, DataArray actualOutput)
        {
            Args.Requires(() => input, () => input != null);
            Args.Requires(() => desiredOutput, () => desiredOutput != null);
            Args.Requires(() => actualOutput, () => actualOutput != null);
            Args.Requires(() => actualOutput, () => desiredOutput.Size == actualOutput.Size);

            Input = input;
            DesiredOutput = desiredOutput;
            ActualOutput = actualOutput;
        }

        public DataArray Input { get; private set; }

        public DataArray DesiredOutput { get; private set; }

        public DataArray ActualOutput { get; private set; }

        public bool HasOutput
        {
            get { return DesiredOutput != null && ActualOutput != null; }
        }

        bool disposed;

        public void Dispose()
        {
            if (!disposed)
            {
                disposed = true;

                Input.Dispose();
                if (DesiredOutput != null) DesiredOutput.Dispose();
                if (ActualOutput != null) ActualOutput.Dispose();

                GC.SuppressFinalize(this);
            }
        }
    }
}
